browser/brave_shields/https_upgrade_browsertest.cc

```diff
@@ -0,0 +1,227 @@
+    brave_shields::SetBraveShieldsEnabled(ContentSettings(), shields_enabled,
+                                          initial_url, nullptr);
+    brave_shields::SetHttpsUpgradeControlType(
+        ContentSettings(), test_case.control_type,
+        global_setting ? GURL() : initial_url,
+        g_browser_process->local_state());
+    AttemptToNavigateToURL(initial_url);
+    return initial_url;
+  }
+
+  content::WebContents* Contents() {
+    return chrome_test_utils::GetActiveWebContents(this);
+  }
+
+  HostContentSettingsMap* ContentSettings() {
+    return HostContentSettingsMapFactory::GetForProfile(
+        chrome_test_utils::GetProfile(this));
+  }
+
+ protected:
+  base::test::ScopedFeatureList feature_list_;
+  net::EmbeddedTestServer* http_server() { return &http_server_; }
+  net::EmbeddedTestServer* https_server() { return &https_server_; }
+
+ private:
+  net::EmbeddedTestServer http_server_{net::EmbeddedTestServer::TYPE_HTTP};
+  net::EmbeddedTestServer https_server_{net::EmbeddedTestServer::TYPE_HTTPS};
+  content::ContentMockCertVerifier mock_cert_verifier_;
+};
+
+class HttpsUpgradeBrowserTest_FlagDisabled : public HttpsUpgradeBrowserTest {
+ public:
+  HttpsUpgradeBrowserTest_FlagDisabled() = default;
+  ~HttpsUpgradeBrowserTest_FlagDisabled() override = default;
+
+  void SetUp() override {
+    feature_list_.InitAndDisableFeature(kBraveHttpsByDefault);
+    PlatformBrowserTest::SetUp();
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(HttpsUpgradeBrowserTest, CheckUpgrades) {
+  for (bool global_setting : {true, false}) {
+    for (const TestCase& test_case : kTestCases) {
+      RunTestCaseNavigation(true, global_setting, test_case);
+      bool interstitial_showing =
+          chrome_browser_interstitials::IsShowingInterstitial(Contents());
+      if (test_case.expected_result == PageResult::kInterstitial) {
+        EXPECT_TRUE(interstitial_showing);
+      } else {
+        EXPECT_FALSE(interstitial_showing);
+        GURL final_url =
+            (test_case.expected_result == PageResult::kHttp ? http_server()
+                                                            : https_server())
+                ->GetURL(test_case.domain, "/simple.html");
+        EXPECT_EQ(final_url, Contents()->GetLastCommittedURL());
+      }
+    }
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(HttpsUpgradeBrowserTest, CheckUpgradesWithShieldsDown) {
+  for (bool global_setting : {true, false}) {
+    for (const TestCase& test_case : kTestCases) {
+      const GURL initial_url =
+          RunTestCaseNavigation(false, global_setting, test_case);
+      // Shields down means no URLs change and no interstitials shown.
+      EXPECT_EQ(initial_url, Contents()->GetLastCommittedURL());
+      bool interstitial_showing =
+          chrome_browser_interstitials::IsShowingInterstitial(Contents());
+      EXPECT_FALSE(interstitial_showing);
+    }
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(HttpsUpgradeBrowserTest_FlagDisabled, CheckUpgrades) {
```

Comment:
nit: I'd move the `FlagDisabled` bit to the `CheckUpgrades` side so that all browser tests for this feature can be run using the `HttpsUpgradeBrowserTest.*` filter

Reply:
Because we are using a separate class for the disabled case, it unfortunately uses this awkward name. But we can run all 3 tests by using the filter `HttpsUpgradeBrowserTest*` (note the absence of a dot).